### ISSUE TITLE
Permissions error on startup should be made optionally silent

### ISSUE THREAD

**Issue by bittylicious** [pwr-Solaar/Solaar#465]
It should be possible to make the permissions error silent.

This is the error that says, "Found a Logitech received (...), but did not have permission to open it".

On occasion, this is intentional. For instance here, we have two seats controlled by one computer, so two unifying receivers (one for each seat) and a different user usually accessing each one.

Having the error pop up on each login is ugly. I do agree it should appear by default, or perhaps it should only appear if there are no receivers detected (rather than one receiver detected).

**Comment by pfps** [comment]
Is the problem that this error has to be responded to?  It might be possible to turn the error pop-up into a notification.

**Comment by bittylicious** [comment]
Yes, sort of. It is worse that it needs to be responded to, but in a multi-seat situation, it is intentional that Solaar only has access to one unifying receiver. My situation has two seats, with each seat having access to only its own USB controllers. There is one unifying receiver for each seat.

In my situation, when seat 2 starts using the mouse (for instance), the pop up notification pops up unexpectedly on seat 1. This happens every time after being awoken after the mouse goes to sleep.

I think there should be an option to ignore situations where a receiver gets detected but the current Solaar instance doesn't have access to it. Even a notification wouldn't be great, as that would still appear on the other user's seat screen.

**Comment by pfps** [comment]
The solution is, I think, to look at the ACLs to see if the device is for another user.  However, there appears to be a problem with pylibacl - trying to install it produces an error.
So the right fix isn't available.  
Also, receivers are processed in an undetermined order so only producing an error if there is no other receiver will be difficult.

**Comment by bittylicious** [comment]
To the ACLs show what seat a device is allocated to? I believe that I can log in as the same user on both seats, but Solaar still won't be able to access the other device (hence looking at user ACLs wouldn't work).

It's not a perfect fix, but could the error be suppressed if Solaar is able to correctly connect to a receiver already, i.e. not error if it doesn't have access to a second one that just "appeared" from its perspective?

**Comment by pfps** [comment]
I hacked together something that appears to do what you are asking for.  Unfortunately it uses pylibacl, which requires two development packages.  As well pylibacl appears to be very alpha, so I'm not confident that it is something that should be depended on.

The real way to do this would be via libinput, I think, but that appears to be hard to navigate.

So, you could try cloning the source and making the following change to the end of lib/solaar/listener.py

```
		except OSError:
			# permission error, ignore this path for now
##sudo dnf install python3-devel libacl-devel
##pip install --user pylibacl
			import posix1e
			# if the receiver has extended ACL then it is likely for another seat
			if not posix1e.has_extended(device_info.path):
				_error_callback('permissions', device_info.path)
```

The comment lines are what you need to do to install pylibacl (on Fedora - you may have to change the packages on other distributions).

**Comment by pfps** [comment]
The ACLs show the user, not the seat.   If you are logged in to two seats you should be able to control both receivers.

**Comment by bittylicious** [comment]
Your hack seemed to work! I tried MASTER before (it gave the error) and afterwards (no error, regular access to my device). Thank you very much.

Obviously though, it does introduce extra dependencies for this somewhat niche use case. I'm obviously fine with this, but I'm not sure other users would be. Without posix1e pip installed, I get a Python warning at the import stage, as expected.

Interesting about ACLs - I had always assumed seats were fully independent and that udev restricted access solely based on seat, not user. I probably am wrong though.

**Comment by pfps** [comment]
I have a change that calls getfacl to determine whether there are any ACLs for the receiver.  This is a hack, but I think better than adding dependencies for this case.

**Comment by pfps** [comment]
Fixed by #769
